osu.Desktop/HighPerformanceSession/Gamemode.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Runtime.InteropServices;
+
+namespace Gamemode
+{
+    public static class GamemodeRequest
+    {
+
+        [DllImport("libgamemode.so.0")]
+        private static extern int real_gamemode_request_start();
+
+        [DllImport("libgamemode.so.0")]
+        private static extern int real_gamemode_request_end();
+
+        public static int RequestStart()
+        {
+            try
+            {
+                return real_gamemode_request_start();
+            }
+            catch
+            {
+                return -1;
+            }
+        }
```

Comment:
not sure I care for the C-style API of returning `int` error codes, nor for the silencing of exceptions in the `catch` branch and coalescing them to an information-obscuring error code instead of logging properly. in my view all exception/error code handling related to this `libgamemode` library should be done locally in this class.